src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,110 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTORY = 0.75;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTORY * DEFAULT_INITIAL_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> node = new Node<>(key, value, getIndex(key));
+        if (needToResize()) {
+            resize();
+        }
+        if (table[node.hash] == null) {
+            table[node.hash] = node;
+            size++;
+        } else {
+            Node oldNode = table[node.hash];
+            while (oldNode.next != null) {
+                if (checkEquals(oldNode.key, key)) {
+                    oldNode.value = node.value;
+                    return;
+                }
+                oldNode = oldNode.next;
+            }
```

Comment:
repeating code in put and getValue methods
```suggestion
            while (oldNode.next != null) {
                if (checkEquals(oldNode.key, key)) {
                    oldNode.value = node.value;
                    return;
                }
                oldNode = oldNode.next;
            }
```